eventdemo/source/android/app/src/main/java/org/kaaproject/kaa/demo/events/utils/KaaChatManager.java

```diff
@@ -0,0 +1,285 @@
+import org.kaaproject.kaa.client.KaaClient;
+import org.kaaproject.kaa.client.KaaClientProperties;
+import org.kaaproject.kaa.client.SimpleKaaClientStateListener;
+import org.kaaproject.kaa.client.event.EventFamilyFactory;
+import org.kaaproject.kaa.client.event.registration.UserAttachCallback;
+import org.kaaproject.kaa.common.endpoint.gen.SyncResponseResultType;
+import org.kaaproject.kaa.common.endpoint.gen.UserAttachResponse;
+import org.kaaproject.kaa.examples.event.Chat;
+import org.kaaproject.kaa.examples.event.ChatEvent;
+import org.kaaproject.kaa.examples.event.ChatEventType;
+import org.kaaproject.kaa.examples.event.Message;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * Modified code from the Java event demo
+ */
+public final class KaaChatManager {
+
+    private static final String KEYS_DIR = "keys_for_java_event_demo";
+
+    private final Context mContext;
+    private final List<String> mChatList = new ArrayList<>();
+
+    // default rooms
+    {
+        mChatList.add("Living room");
+        mChatList.add("Guest room");
+    }
+
+    // to send events on main thread
+    private final Handler mMainThreadHandler = new Handler(Looper.getMainLooper());
+
+    private KaaClient mKaaClient;
+    private Chat mChatEventFamily;
+
+    private final List<Chat.Listener> mChatListeners = new ArrayList<>();
+
+    public KaaChatManager(Context ctx) {
+        mContext = ctx;
+    }
+
+    /**
+     * Startup current endpoint and init event listeners for receiving chat user messages and
+     * events to manage chat list
+     *
+     * @throws IOException
+     */
+    public void start() throws IOException {
+        try {
+
+            // Setup working directory for endpoint
+            final KaaClientProperties endpointProperties = new KaaClientProperties();
+            endpointProperties.setWorkingDirectory(KEYS_DIR);
+
+            // Create the Kaa desktop context for the application
+            final AndroidKaaPlatformContext androidKaaPlatformContext =
+                    new AndroidKaaPlatformContext(mContext, endpointProperties);
+
+            // Create a Kaa client and add a listener which creates a log record
+            // as soon as the Kaa client is started.
+            final CountDownLatch startupLatch = new CountDownLatch(1);
+            mKaaClient = Kaa.newClient(androidKaaPlatformContext, new SimpleKaaClientStateListener() {
+                @Override
+                public void onStarted() {
+                    toast("Kaa client started");
+                    startupLatch.countDown();
+                }
+
+                @Override
+                public void onStopped() {
+                    toast("Kaa client stopped");
+                }
+            }, true);
+
+            //Start the Kaa client and connect it to the Kaa server.
+            mKaaClient.start();
+
+            startupLatch.await();
```

Comment:
It's also stopping EventDemoApp from being created while you're waiting for result.